server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -463,16 +463,23 @@ public XContentBuilder snapshotsToXContent(final XContentBuilder builder, final
         for (final SnapshotId snapshot : getSnapshotIds()) {
             builder.startObject();
             builder.field(NAME, snapshot.getName());
-            builder.field(UUID, snapshot.getUUID());
-            if (snapshotStates.containsKey(snapshot.getUUID())) {
-                builder.field(STATE, snapshotStates.get(snapshot.getUUID()).value());
+            final String snapshotUUID = snapshot.getUUID();
+            builder.field(UUID, snapshotUUID);
+            final SnapshotState state = snapshotStates.get(snapshotUUID);
+            if (state != null) {
+                builder.field(STATE, state.value());
             }
             if (shouldWriteIndexGens) {
-                builder.field(INDEX_METADATA_LOOKUP, indexMetaDataGenerations.lookup.getOrDefault(snapshot, Collections.emptyMap())
-                    .entrySet().stream().collect(Collectors.toMap(entry -> entry.getKey().getId(), Map.Entry::getValue)));
```

Comment:
No need to instantiate a whole new map here in a loop for each snapshot when we can just extract the index id in a loop as we go.